application/common/models/User.php

```diff
@@ -343,6 +343,55 @@ private function savePassword()
         return true;
     }
 
+    public static function getExpiringUsers($criteria): array
+    {
+        if (empty($criteria)) {
+            return [];
+        }
+
+        $users = User::find()->joinWith('currentPassword')
+                             ->where(['active' => 'yes']);
+
+        foreach ($criteria as $name => $value) {
+            switch ($name) {
+                case 'expires_on':
+                case 'grace_period_ends_on':
+                    $users->andWhere(["password.$name" => $value]);
+                    break;
+                default:
+                    // if no criteria names match, this will ensure an empty result is returned
+                    $users->where('0=1');
+            }
+        }
+
+        return $users->all();
+    }
+
+    public static function getNewUsers($createdOn): array
```

Comment:
This looks like it will return the users whose first password was created on the specified date (not necessarily users that we added on the specified date). Was that the intent? Since a user record might not have a password until long after the user record is added to ID Broker (such as if the user doesn't log in for quite some time), it seemed like the name and the actual effect didn't quite line up for this function.